lib/service/mock_api_service.dart

```diff
@@ -0,0 +1,19 @@
+import 'dart:math';
+
+import 'package:dartz/dartz.dart';
+import 'package:injectable/injectable.dart';
+
+@LazySingleton()
+class MockApiService {
+
+  List<String> get items => [for (int i = 0; i < 9999999; i++) 'My List Item nr $i'];
```

Comment:
Hey, I think your test size of 9999999 here is quite large and has a strong tendency to cause latency issues, I would suggest we apply some form of pagination to the data source
![Screenshot_20220210-232134_NetguruCR](https://user-images.githubusercontent.com/8009256/153506768-fca0e20f-3ccb-4bc2-b4f8-c88035836cc4.jpg)